CHANGELOG.md

```diff
@@ -2,6 +2,85 @@
 
 This file documents all notable changes to Juttle. The release numbering uses [semantic versioning](http://semver.org).
 
+## 0.5.0
+
+### Major Changes
+
+- Implemented consolidated, versioned adapter API. Adapters should use the `AdapterAPI` global variable to access the API instead of `require`-ing modules from the `juttle` package directly. [[#367](https://github.com/juttle/juttle/issues/367)]
+
+### Minor Changes
+
+- Made `reduce -every` emit marks at the end of epoch. [[#52](https://github.com/juttle/juttle/issues/52)]
+- Exposed the list of configured adapters in Juttle runtime (`Juttle.adapters()`) and CLI (`juttle --adapters`). [[#313](https://github.com/juttle/juttle/issues/313)]
+- Added the `-append` option to `write file`. It currently works only with CSV nad JSONL formats. [[#286](https://github.com/juttle/juttle/issues/286)]
+- Added support for time range options to `read http`. [[#194](https://github.com/juttle/juttle/issues/194)]
+- Made debug logging controllable using the `DEBUG` environment variable. [[#373](https://github.com/juttle/juttle/issues/373)]
+- Removed `RT-` and `JUTTLE-` prefixes from error codes. Also removed the `-ERROR` suffix in cases where it was redundant. [[#356](https://github.com/juttle/juttle/pull/356)]
+- Disallowed `-from` to be equal to `-to` in sources. [[#322](https://github.com/juttle/juttle/issues/322)]
+- Cleaned up proc inheritance chain. [[#371](https://github.com/juttle/juttle/pull/371)]
+- Added allowed option validation logic into `base` and used it in built-in procs. [[#317](https://github.com/juttle/juttle/pull/317)]
+- Added required option validation logic into `base`. [[#353](https://github.com/juttle/juttle/pull/353)]
+- Changed `procName` from a prototype property on the procs into a function. [[#364](https://github.com/juttle/juttle/pull/364)]
+- Unique proc identifier (`pname`) is now passed to procs in `params` which are now also attached to proc instances. [[#365](https://github.com/juttle/juttle/pull/365)]
+- In views, changed `name` from a prototype property to an instance variable. [[#375](https://github.com/juttle/juttle/pull/375)]
```

Comment:
I consider adapter creators and embedders as users too, so I also included entries relevant for them. There is certainly some fuzzy boundary there because we don’t document the API, so it was sometimes hard to say tell whether a change was just an internal thing or not.

I think I’ll just keep the list. For future releases, we may consider structuring the changelog differently (sections like “Language”, “Adapters”, “API”, etc.) and define the boundary more explicitly.